### PR TITLE
stream: reworked batch consumer;

### PR DESCRIPTION
**Pull request by applike-ss**

### CODE REVIEW

**Review by selm0 — CHANGES_REQUESTED** [review]
Please unify the consumer and batch consumer by adding a base consumer.
Adhere to DRY.

File: pkg/stream/base_consumer.go [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+func (c *baseConsumer) recover() {
+	err := coffin.ResolveRecovery(recover())
+	if err == nil {
+		return
+	}
+
+	c.logger.Error(err, err.Error())
+}
+
+func (c *baseConsumer) callBatchCallback(ctx context.Context, batch []*Message, models []interface{}, attributes []map[string]interface{}) int {
```
Comment (selm0) [inline pkg/stream/base_consumer.go:320]: The base consumer must not know about the consumption process, i.e. whether it's batch or not.
Those are implementation details of the actual consumers.
Please move anything related to that to the specific implementation.

Consider adding a new consumer type which does not fit the batch or single processing pattern, then you would need to touch the base consumer again.

Comment (selm0) [reply pkg/stream/base_consumer.go:320]: I think this requires moving of `doConsuming` aswell

**Review by selm0 — CHANGES_REQUESTED** [review]

File: pkg/stream/base_consumer.go (hunk 1 above)
Comment (applike-ss) [reply pkg/stream/base_consumer.go:320]: This is what we had before you told me to migrate the two. Maybe you want to finish the implementation yourself?

**Review by j4k4 — APPROVED** [review]
+1

**Review by selm0 — APPROVED** [review]

**Pull request merged**